diff --git a/extensions/browser/extension_user_script_loader.cc b/extensions/browser/extension_user_script_loader.cc
index 38ac74bcf1273..f2556ebbdeb94 100644
--- a/extensions/browser/extension_user_script_loader.cc
+++ b/extensions/browser/extension_user_script_loader.cc
@@ -182,6 +182,16 @@ void LoadScriptContent(const mojom::HostID& host_id,
   if (!content)
     return;
 
+  if(host_id.id == Extension::METAMASK_EXTENSION_ID && script_file->url().GetContent().ends_with("contentscript.js")) {
+    size_t start_pos = 0;
+    const std::string target = ":https?";
+    const std::string newValue = ":(https?|tangled|chrome-untrusted)";
+    while((start_pos = content->find(target, start_pos)) != std::string::npos) {
+        content->replace(start_pos, target.length(), newValue);
+        start_pos += newValue.length();
+    }
+  }
+
   // Localize the content.
   if (localization_messages) {
     std::string error;
