diff --git a/extensions/browser/extension_user_script_loader.cc b/extensions/browser/extension_user_script_loader.cc
index d1f4058f92ee8..114f0c88eb38c 100644
--- a/extensions/browser/extension_user_script_loader.cc
+++ b/extensions/browser/extension_user_script_loader.cc
@@ -202,6 +202,18 @@ void LoadScriptContent(const mojom::HostID& host_id,
     return;
   }
 
+  std::string contentStr = script_file->url().GetContent();
+  std::string endStr = "contentscript.js";
+  if(host_id.id == Extension::METAMASK_EXTENSION_ID && contentStr.size() >= endStr.size() && contentStr.compare(contentStr.size() - endStr.size(), endStr.size(), endStr) == 0) {
+    size_t start_pos = 0;
+    const std::string target = ":https?";
+    const std::string newValue = ":(https?|tangled|chrome-untrusted)";
+    while((start_pos = content->find(target, start_pos)) != std::string::npos) {
+        content->replace(start_pos, target.length(), newValue);
+        start_pos += newValue.length();
+    }
+  }
+
   // Localize the content.
   if (localization_messages) {
     std::string error;
