diff --git a/components/os_crypt/sync/key_storage_linux.cc b/components/os_crypt/sync/key_storage_linux.cc
index d9e06d9b901c9..1594f26ac7936 100644
--- a/components/os_crypt/sync/key_storage_linux.cc
+++ b/components/os_crypt/sync/key_storage_linux.cc
@@ -29,8 +29,8 @@
 const char KeyStorageLinux::kFolderName[] = "Chrome Keys";
 const char KeyStorageLinux::kKey[] = "Chrome Safe Storage";
 #else
-const char KeyStorageLinux::kFolderName[] = "Chromium Keys";
-const char KeyStorageLinux::kKey[] = "Chromium Safe Storage";
+const char KeyStorageLinux::kFolderName[] = "Tangled Keys";
+const char KeyStorageLinux::kKey[] = "Tangled Safe Storage";
 #endif
 
 namespace {
@@ -143,7 +143,7 @@ std::unique_ptr<KeyStorageLinux> KeyStorageLinux::CreateServiceInternal(
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
   static const base::NoDestructor<std::string> kDefaultApplicationName("chrome");
 #else
-  static const base::NoDestructor<std::string> kDefaultApplicationName("chromium");
+  static const base::NoDestructor<std::string> kDefaultApplicationName("tangled");
 #endif
 
   std::unique_ptr<KeyStorageLinux> key_storage;
