diff --git a/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc b/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
index b1846840f938f..4679ac4c3d8e4 100644
--- a/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
+++ b/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
@@ -108,2 +108,4 @@
 #include "url/gurl.h"
+#include "chrome/browser/ui/webui/millix/millix_bar.h"
+#include "chrome/browser/ui/webui/millix/millix_app.h"
 
@@ -849,6 +849,10 @@ WebUIFactoryFunction GetWebUIFactoryFunction(WebUI* web_ui,
     return &NewWebUI<IdentityInternalsUI>;
   if (url.host_piece() == chrome::kChromeUINewTabHost)
     return &NewWebUI<NewTabUI>;
+  if (url.host_piece() == chrome::kChromeUIMillixBarHost)
+    return &NewWebUI<MillixBarUI>;
+  if (url.host_piece() == chrome::kChromeUIMillixAppHost)
+    return &NewWebUI<MillixAppUI>;
   if (!profile->IsOffTheRecord()) {
     if (url.host_piece() == chrome::kChromeUINewTabPageHost)
       return &NewWebUI<NewTabPageUI>;
