diff --git a/chrome/common/channel_info_posix.cc b/chrome/common/channel_info_posix.cc
index 5cd53f84f44bb..b5823e71973f8 100644
--- a/chrome/common/channel_info_posix.cc
+++ b/chrome/common/channel_info_posix.cc
@@ -120,7 +120,7 @@ std::string GetDesktopName(base::Environment* env) {
   std::string name;
   if (env->GetVar("CHROME_DESKTOP", &name) && !name.empty())
     return name;
-  return "chromium-browser.desktop";
+  return "tangled-browser.desktop";
 #endif
 }
 #endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS_LACROS)
