diff --git a/chrome/common/channel_info_posix.cc b/chrome/common/channel_info_posix.cc
index f6bcb3cc8e8d6..20cc6ca889d1d 100644
--- a/chrome/common/channel_info_posix.cc
+++ b/chrome/common/channel_info_posix.cc
@@ -148,7 +148,7 @@ std::string GetDesktopName(base::Environment* env) {
   if (name.has_value() && !name.value().empty()) {
     return name.value();
   }
-  return "chromium-browser.desktop";
+  return "tangled-browser.desktop";
 #endif
 }
 #endif  // BUILDFLAG(IS_LINUX)
