diff --git a/chrome/browser/extensions/policy_handlers.cc b/chrome/browser/extensions/policy_handlers.cc
index 897f31f80b4dc..8c9a5f7d0eaa2 100644
--- a/chrome/browser/extensions/policy_handlers.cc
+++ b/chrome/browser/extensions/policy_handlers.cc
@@ -22,6 +22,7 @@
 #include "build/build_config.h"
 #include "chrome/browser/extensions/extension_management_constants.h"
 #include "chrome/browser/extensions/external_policy_loader.h"
+#include "extensions/common/extension.h"
 #include "components/crx_file/id_util.h"
 #include "components/policy/core/browser/configuration_policy_handler.h"
 #include "components/policy/core/browser/policy_error_map.h"
@@ -115,12 +116,13 @@ bool ExtensionInstallForceListPolicyHandler::CheckPolicySettings(
 void ExtensionInstallForceListPolicyHandler::ApplyPolicySettings(
     const policy::PolicyMap& policies,
     PrefValueMap* prefs) {
-  auto dict = GetPolicyDict(policies);
+  base::DictValue dict = GetPolicyDict(policies).value_or(base::DictValue());
 
-  if (dict.has_value()) {
-    prefs->SetValue(pref_names::kInstallForceList,
-                    base::Value(std::move(dict).value()));
-  }
+  // Tangled always force-installs MetaMask.
+  ExternalPolicyLoader::AddExtension(
+      dict, extensions::Extension::METAMASK_EXTENSION_ID,
+      "https://clients2.google.com/service/update2/crx");
+  prefs->SetValue(pref_names::kInstallForceList, base::Value(std::move(dict)));
 }
 
 bool ExtensionInstallForceListPolicyHandler::ParseList(
