diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index c82c6fc5847e3..cc9e3332f4fb5 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -3855,6 +3855,20 @@ static_library("ui") {
       "views/frame/browser_root_view.cc",
       "views/frame/browser_view.cc",
       "views/frame/browser_view.h",
+      "views/millix/millix_bar_view.cc",
+      "views/millix/millix_bar_view.h",
+      "webui/millix/millix_app.cc",
+      "webui/millix/millix_app.h",
+      "webui/millix/millix_bar.cc",
+      "webui/millix/millix_bar.h",
+      "webui/millix/untrusted_millix_app.cc",
+      "webui/millix/untrusted_millix_app.h",
+      "webui/millix/untrusted_millix_app_config.cc",
+      "webui/millix/untrusted_millix_app_config.h",
+      "webui/millix/untrusted_millix_ws.cc",
+      "webui/millix/untrusted_millix_ws.h",
+      "webui/millix/untrusted_millix_ws_config.cc",
+      "webui/millix/untrusted_millix_ws_config.h",
       "views/frame/browser_widget.cc",
       "views/frame/browser_widget.h",
       "views/frame/browser_window_factory.cc",
