diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index d6686618cb86b..0682f218e188e 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -86,12 +86,12 @@ if (is_win) {
     # initialexe/ is used so that the we can reorder imports and write back to
     # the final destination at $root_out_dir/.
     inputs = [
-      "$root_out_dir/initialexe/chrome.exe",
-      "$root_out_dir/initialexe/chrome.exe.pdb",
+      "$root_out_dir/initialexe/tangled.exe",
+      "$root_out_dir/initialexe/tangled.exe.pdb",
     ]
     outputs = [
-      "$root_out_dir/chrome.exe",
-      "$root_out_dir/chrome.exe.pdb",
+      "$root_out_dir/tangled.exe",
+      "$root_out_dir/tangled.exe.pdb",
     ]
     args = [
       "-i",
@@ -149,9 +149,9 @@ if (!is_android && !is_mac) {
   }
 
   if (is_win) {
-    _chrome_output_name = "initialexe/chrome"
+    _chrome_output_name = "initialexe/tangled"
   } else {
-    _chrome_output_name = "chrome"
+    _chrome_output_name = "tangled"
   }
 
   executable("chrome_initial") {
@@ -410,7 +410,7 @@ if (is_win) {
       "app/startup_timestamps.h",
     ]
 
-    output_name = "chrome"
+    output_name = "tangled"
 
     deps = [
       ":chrome_dll_manifest",
@@ -1668,7 +1668,7 @@ if (is_android) {
 if (is_win && enable_resource_allowlist_generation) {
   generate_resource_allowlist("resource_allowlist") {
     deps = [ ":chrome_dll" ]
-    inputs = [ "$root_out_dir/chrome.dll.pdb" ]
+    inputs = [ "$root_out_dir/tangled.dll.pdb" ]
     output = _chrome_resource_allowlist
   }
 }
