diff --git a/tools/polymer/html_to_wrapper.gni b/tools/polymer/html_to_wrapper.gni
index 37a146356b772..74c30c9eb457c 100644
--- a/tools/polymer/html_to_wrapper.gni
+++ b/tools/polymer/html_to_wrapper.gni
@@ -74,7 +74,8 @@ template("html_to_wrapper") {
     }
 
     if (defined(invoker.scheme)) {
-      assert(invoker.scheme == "chrome" || invoker.scheme == "relative")
+      assert(invoker.scheme == "chrome" || invoker.scheme == "tangled" ||
+             invoker.scheme == "relative")
       args += [
         "--scheme",
         invoker.scheme,
