diff --git a/tools/polymer/html_to_wrapper.py b/tools/polymer/html_to_wrapper.py
index 4a361d517cfbe..e6daa50e1f325 100644
--- a/tools/polymer/html_to_wrapper.py
+++ b/tools/polymer/html_to_wrapper.py
@@ -159,7 +159,7 @@ def main(argv):
                       choices=['polymer', 'lit', 'native', 'detect'],
                       default='lit')
   parser.add_argument('--scheme',
-                      choices=['chrome', 'relative'],
+                      choices=['chrome', 'tangled', 'relative'],
                       default='relative')
 
   args = parser.parse_args(argv)
@@ -229,7 +229,7 @@ def main(argv):
 
       substitutions = {
           'content': html_content,
-          'scheme': 'chrome:' if args.scheme == 'chrome' else '',
+          'scheme': 'tangled:' if args.scheme in ['chrome', 'tangled'] else '',
       }
 
       if template_type == 'lit_icons':
