diff --git a/chrome/installer/mini_installer/BUILD.gn b/chrome/installer/mini_installer/BUILD.gn
index 5f9acdb7f3550..25c29ecee8369 100644
--- a/chrome/installer/mini_installer/BUILD.gn
+++ b/chrome/installer/mini_installer/BUILD.gn
@@ -145,9 +145,9 @@ action("mini_installer_archive") {
   release_file = "chrome.release"
 
   inputs = [
-    "$root_out_dir/chrome.dll",
-    "$root_out_dir/chrome_elf.dll",
-    "$root_out_dir/chrome.exe",
+    "$root_out_dir/tangled.dll",
+    "$root_out_dir/tangled_elf.dll",
+    "$root_out_dir/tangled.exe",
     "$root_out_dir/locales/en-US.pak",
     "$root_out_dir/setup.exe",
     "//chrome/tools/build/win/makecab.py",
