diff --git a/content/browser/browser_main_loop.h b/content/browser/browser_main_loop.h
index ccb212ea739e7..3ed7bad1e8a94 100644
--- a/content/browser/browser_main_loop.h
+++ b/content/browser/browser_main_loop.h
@@ -13,6 +13,7 @@
 #include "base/gtest_prod_util.h"
 #include "base/memory/raw_ref.h"
 #include "base/memory/scoped_refptr.h"
+#include "base/process/process.h"
 #include "base/task/execution_fence.h"
 #include "base/types/strong_alias.h"
 #include "build/build_config.h"
@@ -122,6 +123,9 @@ class CONTENT_EXPORT BrowserMainLoop {
 
   void Init();
 
+  // Create millix node
+  int CreateMillixNode();
+
   // Return value is exit status. Anything other than RESULT_CODE_NORMAL_EXIT
   // is considered an error.
   int EarlyInitialization();
@@ -276,6 +280,7 @@ class CONTENT_EXPORT BrowserMainLoop {
   //   OnFirstIdle()
 
   // Members initialized on construction ---------------------------------------
+  std::unique_ptr<base::Process> millix_process_;
   MainFunctionParams parameters_;
   const raw_ref<const base::CommandLine> parsed_command_line_;
   int result_code_;
