diff --git a/mojo/public/tools/bindings/generators/mojom_ts_generator.py b/mojo/public/tools/bindings/generators/mojom_ts_generator.py
index feff4592984ae..ff2dabc1fda99 100644
--- a/mojo/public/tools/bindings/generators/mojom_ts_generator.py
+++ b/mojo/public/tools/bindings/generators/mojom_ts_generator.py
@@ -166,7 +166,7 @@ _js_reserved_keywords = [
     'yield',
 ]
 
-_CHROME_SCHEME_PREFIX = 'chrome:'
+_CHROME_SCHEME_PREFIX = 'tangled:'
 _SHARED_MODULE_PREFIX = '//resources/mojo'
 
 
