diff --git a/chrome/chrome_elf/BUILD.gn b/chrome/chrome_elf/BUILD.gn
index 9d96774f5f3de..dc02359f66fb4 100644
--- a/chrome/chrome_elf/BUILD.gn
+++ b/chrome/chrome_elf/BUILD.gn
@@ -55,6 +55,9 @@ shared_library("chrome_elf") {
     "chrome_elf_main.cc",
     "chrome_elf_main.h",
   ]
+
+  output_name = "tangled_elf"
+
   if (target_cpu == "x86") {
     sources += [ "chrome_elf_x86.def" ]
   } else if (target_cpu == "arm64") {
