diff --git a/chrome/chrome_elf/chrome_elf_x64.def b/chrome/chrome_elf/chrome_elf_x64.def
index cdc6f52fb8461..c8328d0f83e33 100644
--- a/chrome/chrome_elf/chrome_elf_x64.def
+++ b/chrome/chrome_elf/chrome_elf_x64.def
@@ -1,7 +1,7 @@
 ; Copyright 2013 The Chromium Authors. All rights reserved.
 ; Use of this source code is governed by a BSD-style license that can be
 ; found in the LICENSE file.
-LIBRARY  "chrome_elf.dll"
+LIBRARY  "tangled_elf.dll"
 
 EXPORTS
   ; When functions are added to this file, they must also be added to
