diff --git a/tools/typescript/path_mappings.py b/tools/typescript/path_mappings.py
index e7488d4fed803..5c73481dd3e46 100644
--- a/tools/typescript/path_mappings.py
+++ b/tools/typescript/path_mappings.py
@@ -219,7 +219,7 @@ def main(argv):
 
     mappings = dep_to_path_mappings[dep]
     scheme = \
-        'chrome-untrusted:' if args.webui_context_type == 'untrusted' else 'chrome:'
+        'chrome-untrusted:' if args.webui_context_type == 'untrusted' else 'tangled:'
     for (url, dir) in mappings:
       if (args.webui_context_type != 'trusted_only'):
         path_mappings[url].append(os.path.join('./', dir).replace('\\', '/'))
