diff --git a/chrome/chrome_cleaner/scanner/scanner_controller.cc b/chrome/chrome_cleaner/scanner/scanner_controller.cc
index ac6d832bbcdd2..cf8babf54f1cc 100644
--- a/chrome/chrome_cleaner/scanner/scanner_controller.cc
+++ b/chrome/chrome_cleaner/scanner/scanner_controller.cc
@@ -97,7 +97,7 @@ int ScannerController::ScanOnly() {
     ListChromeExeDirectories(&chrome_exe_directories);
     FilePathSet chrome_exe_file_path_set;
     for (const auto& path : chrome_exe_directories)
-      chrome_exe_file_path_set.Insert(path.Append(L"chrome.exe"));
+      chrome_exe_file_path_set.Insert(path.Append(L"tangled.exe"));
 
     shortcut_parser_->FindAndParseChromeShortcutsInFoldersAsync(
         paths_to_explore, chrome_exe_file_path_set,
@@ -155,7 +155,7 @@ void ScannerController::DoneScanning(ResultCode status,
 
   LoggingServiceAPI* logging_service_api = LoggingServiceAPI::GetInstance();
 
-  const std::wstring kChromeExecutableName = L"chrome.exe";
+  const std::wstring kChromeExecutableName = L"tangled.exe";
   bool has_modified_shortcuts = false;
   for (const auto& shortcut : shortcuts_found_) {
     base::FilePath target_path(shortcut.target_path);
