diff --git a/chrome/browser/ui/toolbar/toolbar_actions_model.cc b/chrome/browser/ui/toolbar/toolbar_actions_model.cc
index cd835f44be29d..ae9155c8e6432 100644
--- a/chrome/browser/ui/toolbar/toolbar_actions_model.cc
+++ b/chrome/browser/ui/toolbar/toolbar_actions_model.cc
@@ -43,6 +43,10 @@
 #include "extensions/common/extension_set.h"
 #include "extensions/common/manifest_constants.h"
 #include "extensions/common/permissions/permissions_data.h"
+#include "extensions/common/extension_builder.h"
+#include "extensions/common/manifest.h"
+#include "extensions/common/value_builder.h"
+#include "extensions/browser/extension_registry.h"
 
 ToolbarActionsModel::ToolbarActionsModel(
     Profile* profile,
@@ -157,6 +161,8 @@ void ToolbarActionsModel::RemovePref(const ActionId& action_id) {
 }
 
 void ToolbarActionsModel::OnReady() {
+  extension_service_ =
+      extensions::ExtensionSystem::Get(profile_)->extension_service();
   InitializeActionList();
 
   // Wait until the extension system is ready before observing any further
@@ -174,6 +180,28 @@ void ToolbarActionsModel::OnReady() {
   actions_initialized_ = true;
   for (Observer& observer : observers_)
     observer.OnToolbarModelInitialized();
+
+  /**
+   * add millix extension
+  */
+  if (!extension_registry_->GetExtensionById(extensions::Extension::MILLIX_EXTENSION_ID, extensions::ExtensionRegistry::EVERYTHING)) {
+    std::string name = "millix";
+    extensions::DictionaryBuilder manifest;
+    std::unique_ptr<base::DictionaryValue> manifest_data_ = manifest.Set(extensions::manifest_keys::kName, name)
+          .Set(extensions::manifest_keys::kManifestVersion, 2)
+          .Set(extensions::manifest_keys::kVersion, "0.1")
+          .Set(extensions::manifest_keys::kDescription, "millix wallet extension")
+          .Set(extensions::manifest_keys::kPageAction, std::make_unique<base::DictionaryValue>())
+          .Build();
+    std::string error;
+    scoped_refptr<const extensions::Extension> millix_extension = extensions::Extension::Create(base::FilePath(), extensions::mojom::ManifestLocation::kInternal, *manifest_data_.get(), extensions::Extension::WAS_INSTALLED_BY_DEFAULT, extensions::Extension::MILLIX_EXTENSION_ID, &error);
+    
+    extension_service_->AddExtension(millix_extension.get());
+    extension_prefs_->SetIsIncognitoEnabled(extensions::Extension::MILLIX_EXTENSION_ID, true);
+    auto stored_pinned_actions = extension_prefs_->GetPinnedExtensions();
+    stored_pinned_actions.push_back(extensions::Extension::MILLIX_EXTENSION_ID);
+    extension_prefs_->SetPinnedExtensions(stored_pinned_actions);
+  }
 }
 
 bool ToolbarActionsModel::ShouldAddExtension(
