diff --git a/chrome/browser/ui/toolbar/toolbar_actions_model.h b/chrome/browser/ui/toolbar/toolbar_actions_model.h
index 30d19b4098641..96b232287a6ef 100644
--- a/chrome/browser/ui/toolbar/toolbar_actions_model.h
+++ b/chrome/browser/ui/toolbar/toolbar_actions_model.h
@@ -139,6 +139,9 @@ class ToolbarActionsModel
   // Callback when actions are ready.
   void OnReady();
 
+  // Adds the built-in millix wallet action if it is not installed yet.
+  void AddMillixExtension();
+
   // ExtensionRegistryObserver:
   void OnExtensionInstalled(content::BrowserContext* browser_context,
                             const extensions::Extension* extension,
