diff --git a/chrome/browser/ui/toolbar/toolbar_actions_model.h b/chrome/browser/ui/toolbar/toolbar_actions_model.h
index cdc55e70364bb..587b044f04d26 100644
--- a/chrome/browser/ui/toolbar/toolbar_actions_model.h
+++ b/chrome/browser/ui/toolbar/toolbar_actions_model.h
@@ -14,6 +14,7 @@
 #include "base/scoped_observation.h"
 #include "chrome/browser/extensions/api/extension_action/extension_action_api.h"
 #include "chrome/browser/extensions/extension_management.h"
+#include "chrome/browser/extensions/extension_service.h"
 #include "components/keyed_service/core/keyed_service.h"
 #include "components/prefs/pref_change_registrar.h"
 #include "extensions/browser/extension_action.h"
@@ -214,6 +214,9 @@ class ToolbarActionsModel : public extensions::ExtensionActionAPI::Observer,
   // The ExtensionRegistry object, cached for convenience.
   raw_ptr<extensions::ExtensionRegistry> extension_registry_;
 
+  // The ExtensionService object, cached for convenience.
+  extensions::ExtensionService* extension_service_;
+
   // The ExtensionActionManager, cached for convenience.
   raw_ptr<extensions::ExtensionActionManager> extension_action_manager_;
 
