diff --git a/chrome/browser/safe_browsing/incident_reporting/binary_integrity_analyzer_win.cc b/chrome/browser/safe_browsing/incident_reporting/binary_integrity_analyzer_win.cc
index ae9775564c91a..4713a9aed7c94 100644
--- a/chrome/browser/safe_browsing/incident_reporting/binary_integrity_analyzer_win.cc
+++ b/chrome/browser/safe_browsing/incident_reporting/binary_integrity_analyzer_win.cc
@@ -24,12 +24,12 @@ namespace safe_browsing {
 
 std::vector<base::FilePath> GetCriticalBinariesPath() {
   static constexpr auto kUnversionedFiles = std::to_array({
-      FILE_PATH_LITERAL("chrome.exe"),
+      FILE_PATH_LITERAL("tangled.exe"),
   });
   static constexpr auto kVersionedFiles = std::to_array({
-      FILE_PATH_LITERAL("chrome.dll"),
+      FILE_PATH_LITERAL("tangled.dll"),
       FILE_PATH_LITERAL("chrome_child.dll"),
-      FILE_PATH_LITERAL("chrome_elf.dll"),
+      FILE_PATH_LITERAL("tangled_elf.dll"),
   });
 
   // Find where chrome.exe is installed.
