diff --git a/chrome/browser/safe_browsing/incident_reporting/environment_data_collection_win.cc b/chrome/browser/safe_browsing/incident_reporting/environment_data_collection_win.cc
index 82536877d18a5..3a696a29f02f1 100644
--- a/chrome/browser/safe_browsing/incident_reporting/environment_data_collection_win.cc
+++ b/chrome/browser/safe_browsing/incident_reporting/environment_data_collection_win.cc
@@ -37,8 +37,8 @@ const REGSAM kKeyReadNoNotify = (KEY_READ) & ~(KEY_NOTIFY);
 
 // The modules on which we will run VerifyModule.
 constexpr std::array<const wchar_t*, 3> kModulesToVerify = {{
-    L"chrome.dll",
-    L"chrome_elf.dll",
+    L"tangled.dll",
+    L"tangled_elf.dll",
     L"ntdll.dll",
 }};
 
