diff --git a/chrome/browser/ui/views/frame/browser_view.h b/chrome/browser/ui/views/frame/browser_view.h
index b37a1c4e7a87b..bfffaa203aea8 100644
--- a/chrome/browser/ui/views/frame/browser_view.h
+++ b/chrome/browser/ui/views/frame/browser_view.h
@@ -60,6 +60,7 @@
 #include "ui/views/widget/widget_delegate.h"
 #include "ui/views/widget/widget_observer.h"
 #include "ui/views/window/client_view.h"
+#include "chrome/browser/ui/views/millix/millix_bar_view.h"
 
 #if BUILDFLAG(IS_CHROMEOS_ASH)
 #include "ui/compositor/throughput_tracker.h"
@@ -109,6 +110,7 @@ enum class Channel;
 namespace views {
 class ExternalFocusTracker;
 class WebView;
+class MillixBarView;
 }
 
 ///////////////////////////////////////////////////////////////////////////////
@@ -165,6 +167,8 @@ class BrowserView : public BrowserWindow,
     return top_controls_slide_controller_.get();
   }
 
+  views::MillixBarView* GetMillixBarView();
+
   void SetDownloadShelfForTest(DownloadShelf* download_shelf);
 
   // This suppresses the slide behaviors of top-controls and so the top controls
@@ -997,6 +1001,8 @@ class BrowserView : public BrowserWindow,
   // |------------------------------------------------------------------|
   // | All infobars (infobar_container_)                                |
   // |------------------------------------------------------------------|
+  // | Millix (millix_bar_view_)                                        |
+  // |------------------------------------------------------------------|
   // | Contents container (contents_container_)                         |
   // |  --------------------------------------------------------------  |
   // |  |  devtools_web_view_                                        |  |
@@ -1012,6 +1018,8 @@ class BrowserView : public BrowserWindow,
   // the top views in immersive fullscreen.
   raw_ptr<TopContainerView, DanglingUntriaged> top_container_ = nullptr;
 
+  views::MillixBarView* millix_bar_view_ = nullptr;
+
   // The view that contains the tabstrip, new tab button, and grab handle space.
   raw_ptr<TabStripRegionView, DanglingUntriaged> tab_strip_region_view_ =
       nullptr;
