diff --git a/chrome/browser/ui/views/frame/browser_view.h b/chrome/browser/ui/views/frame/browser_view.h
index aa0ddbd7eef8b..617126f2081bc 100644
--- a/chrome/browser/ui/views/frame/browser_view.h
+++ b/chrome/browser/ui/views/frame/browser_view.h
@@ -62,6 +62,7 @@
 #include "ui/views/widget/widget_delegate.h"
 #include "ui/views/widget/widget_observer.h"
 #include "ui/views/window/client_view.h"
+#include "chrome/browser/ui/views/millix/millix_bar_view.h"
 
 #if BUILDFLAG(IS_CHROMEOS)
 #include "ui/compositor/compositor_metrics_tracker.h"
@@ -344,6 +345,8 @@ class BrowserView : public BrowserWindow,
   // Returns true if the top UI are visible on screen.
   bool GetTabStripVisible() const;
 
+  views::MillixBarView* GetMillixBarView();
+
   // Returns true if the top UI should be drawn.
   // On macOS, it is possible that the top UI is drawn but hidden.
   bool ShouldDrawTabStrip() const;
@@ -1217,6 +1220,8 @@ class BrowserView : public BrowserWindow,
   // title Label for them here.
   raw_ptr<views::Label> web_app_window_title_ = nullptr;
 
+  raw_ptr<views::MillixBarView> millix_bar_view_ = nullptr;
+
   // The view that contains the tabstrip, new tab button, and grab handle space.
   raw_ptr<HorizontalTabStripRegionView> horizontal_tab_strip_region_view_ =
       nullptr;
