diff --git a/chrome/browser/ui/views/frame/browser_view.cc b/chrome/browser/ui/views/frame/browser_view.cc
index bcebaf44f71c0..837da4ac427c3 100644
--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -958,6 +958,18 @@ BrowserView::BrowserView(Browser* browser)
   top_container_separator_->SetProperty(views::kElementIdentifierKey,
                                         kContentsSeparatorTopEdgeElementId);
 
+  millix_bar_view_ = AddChildView(std::make_unique<views::MillixBarView>(this));
+  millix_bar_view_->SetVisible(false);
+
+  GURL millix_url(chrome::kChromeUIMillixBarURL);
+  WebContents::CreateParams millix_create_params(browser_->profile(), content::SiteInstance::CreateForURL(browser_->profile(), millix_url));
+  millix_create_params.desired_renderer_state = WebContents::CreateParams::kInitializeAndWarmupRendererProcess;
+  std::unique_ptr<WebContents> millix_web_contents(WebContents::Create(millix_create_params));
+
+  content::NavigationController::LoadURLParams params(millix_url);
+  millix_web_contents->GetController().LoadURLWithParams(params);
+  millix_bar_view_->SetWebContents(std::move(millix_web_contents));
+
   contents_container_ = AddChildView(std::move(contents_container));
   set_contents_view(contents_container_);
 
@@ -1121,6 +1133,7 @@ BrowserView::~BrowserView() {
   loading_bar_ = nullptr;
   find_bar_host_view_ = nullptr;
   infobar_container_ = nullptr;
+  millix_bar_view_ = nullptr;
   multi_contents_view_ = nullptr;
   main_shadow_overlay_ = nullptr;
   lens_overlay_view_ = nullptr;
@@ -1421,6 +1434,10 @@ bool BrowserView::IsVerticalTabStripCollapsed() const {
   return false;
 }
 
+views::MillixBarView* BrowserView::GetMillixBarView() {
+  return millix_bar_view_;
+}
+
 bool BrowserView::GetIncognito() const {
   return browser_->GetProfile()->IsIncognitoProfile();
 }
@@ -5143,6 +5160,7 @@ void BrowserView::AddedToWidget() {
   layout_views.projects_panel_container = projects_panel_container_;
   layout_views.toolbar = toolbar_;
   layout_views.infobar_container = infobar_container_;
+  layout_views.millix_bar = millix_bar_view_;
   layout_views.contents_container = contents_container_;
   layout_views.multi_contents_view = multi_contents_view_;
   layout_views.toolbar_height_side_panel = toolbar_height_side_panel_;
