diff --git a/chrome/browser/resources/new_tab_page/BUILD.gn b/chrome/browser/resources/new_tab_page/BUILD.gn
index 9ef1a488f46b5..72d606cefa728 100644
--- a/chrome/browser/resources/new_tab_page/BUILD.gn
+++ b/chrome/browser/resources/new_tab_page/BUILD.gn
@@ -26,6 +26,7 @@ build_webui("build") {
 
   extra_grdp_files = [
     "$target_gen_dir/icons/resources.grdp",
+    "$target_gen_dir/tangled/resources.grdp",
     "$target_gen_dir/modules/authentication/icons/resources.grdp",
     "$target_gen_dir/modules/calendar/icons/resources.grdp",
     "$target_gen_dir/modules/tab_groups/icons/resources.grdp",
@@ -33,6 +34,7 @@ build_webui("build") {
 
   extra_grdp_deps = [
     "icons:build_grdp",
+    "tangled:build_grdp",
     "modules/authentication/icons:build_grdp",
     "modules/calendar/icons:build_grdp",
     "modules/tab_groups/icons:build_grdp",
